evm/generators.go

```diff
@@ -1281,131 +1598,193 @@ func {{.HandlerName}}() *cobra.Command {
+					client, clientErr := NewClient(rpc)
+					if clientErr != nil {
+						return clientErr
+					}
+					chainIDCtx, cancelChainIDCtx := NewChainContext(timeout)
+					defer cancelChainIDCtx()
+					chainID, chainIDErr := client.ChainID(chainIDCtx)
+					if chainIDErr != nil {
+						return chainIDErr
+					}
+					safeApi = "https://safe-client.safe.global/v1/chains/" + chainID.String() + "/transactions/" + safeAddress + "/propose"
+					fmt.Println("--safe-api not specified, using default (", safeApi, ")")
 				}
 
-				gasEstimationCtx, cancelGasEstimationCtx := NewChainContext(timeout)
-				defer cancelGasEstimationCtx()
-
-				gasEstimate, gasEstimateErr := client.EstimateGas(gasEstimationCtx, estimationMessage)
-				if gasEstimateErr != nil {
-					return gasEstimateErr
+				if OperationType(safeOperation).String() == "Unknown" {
+					return fmt.Errorf("--safe-operation must be 0 (Call) or 1 (DelegateCall)")
 				}
-
-				transactionBinary, transactionBinaryErr := transaction.MarshalBinary()
-				if transactionBinaryErr != nil {
-					return transactionBinaryErr
+            }
+
+            {{range .MethodArgs}}
+            {{.PreRunE}}
+            {{- end}}
+
+            return nil
+        },
+        RunE: func(cmd *cobra.Command, args []string) error {
+            client, clientErr := NewClient(rpc)
+            if clientErr != nil {
+                return clientErr
+            }
+
+            key, keyErr := KeyFromFile(keyfile, password)
+            if keyErr != nil {
+                return keyErr
+            }
+
+            chainIDCtx, cancelChainIDCtx := NewChainContext(timeout)
+            defer cancelChainIDCtx()
+            chainID, chainIDErr := client.ChainID(chainIDCtx)
+            if chainIDErr != nil {
+                return chainIDErr
+            }
+
+            transactionOpts, transactionOptsErr := bind.NewKeyedTransactorWithChainID(key.PrivateKey, chainID)
+            if transactionOptsErr != nil {
+                return transactionOptsErr
+            }
+
+            SetTransactionParametersFromArgs(transactionOpts, nonce, value, gasPrice, maxFeePerGas, maxPriorityFeePerGas, gasLimit, simulate)
+
+            contract, contractErr := New{{$structName}}(contractAddress, client)
+            if contractErr != nil {
+                return contractErr
+            }
+
+            session := {{$structName}}TransactorSession{
+                Contract: &contract.{{$structName}}Transactor,
+                TransactOpts: *transactionOpts,
+            }
+
+            if safeAddress != "" {
+                // Generate transaction data
+				abi, err := {{$structName}}MetaData.GetAbi()
+                if err != nil {
+                    return fmt.Errorf("failed to get ABI: %v", err)
+                }
+				// TODO: this is a workaround to match the original method name in the ABI, need to fix to work with every method name
+                packedData, err := abi.Pack("{{ToLowerCamel .MethodName}}",
```

Comment:
This would also be useful for the `--ecode` flag: [issue](https://github.com/moonstream-to/seer/issues/19)

No need to address in this PR, but this strategy applies to that, as well.

Reply:
Just added a comment there for this https://github.com/moonstream-to/seer/issues/19#issuecomment-2383971861